test/zero/tools/fixtures.clj

```diff
@@ -32,3 +40,86 @@
             (->>
              (gcs/list-objects gcs-test-bucket name#)
              (run! (comp delete-test-object :name)))))))
+
+(defn- cloud-db-url
+  "Return NIL or an URL for the CloudSQL instance containing DB-NAME."
+  [environment instance-name db-name]
+  (if-let [gcp (get-in env/stuff [environment :server :project])]
+    (letfn [(postgresql? [{:keys [instanceType name region]}]
+              (when (= [instanceType         name]
+                       ["CLOUD_SQL_INSTANCE" instance-name])
+                (str "jdbc:postgresql://google/" db-name "?useSSL=false"
+                     "&socketFactory="
+                     "com.google.cloud.sql.postgres.SocketFactory"
+                     "&cloudSqlInstance="
+                     (str/join ":" [gcp region name]))))]
+      (-> {:method :get
+           :url (str "https://www.googleapis.com/sql/v1beta4/projects/"
+                     gcp "/instances")
+           :headers (merge {"Content-Type" "application/json"}
+                           (once/get-auth-header))}
+          http/request
+          :body
+          (json/read-str :key-fn keyword)
+          :items
+          (->> (keep postgresql?))
+          first))))
+
+(defn- run-liquibase
+  "Migrate the database schema using Liquibase."
+  [url username password]
+  (let [status (Main/run
+                 (into-array
+                   String
+                   [(str "--url=" url)
+                    (str "--changeLogFile=database/changelog.xml")
+                    (str "--username=" username)
+                    (str "--password=" password)
+                    "update"]))]
+    (when-not (zero? status)
+      (throw
+        (Exception.
+          (format "Liquibase failed with: %s" status))))))
+
+(def testing-dbname
+  (:db-name (postgres/zero-db-config)))
+
+(defn create-db
+  "Create the ad-hoc testing database with DBNAME."
+  [dbname]
+  (if (System/getenv "ZERO_DEPLOY_ENVIRONMENT")
+    ; cloud
+    (util/shell! "gcloud" "sql" "databases" "create" dbname "-i" "zero-postgresql")
+    ; local
+    (util/shell! "createdb" dbname)))
+
+(defn setup-db
+  "Setup te db by running liquibase migrations by DBNAME."
+  [dbname]
+  (if-let [test-env (System/getenv "ZERO_DEPLOY_ENVIRONMENT")]
+    ; cloud
+    (let [url (cloud-db-url :gotc-dev "zero-postgresql" dbname)
+          {:keys [username password]}
+          (-> env/stuff test-env :server :vault util/vault-secrets)]
+      (run-liquibase url username password))
+    ; local
+    (run-liquibase (format "jdbc:postgresql:%s" dbname) (System/getenv "USER") "")))
+
+(defn destroy-db
+  "Tear down the testing database by DBNAME."
+  [dbname]
+  (if (System/getenv "ZERO_DEPLOY_ENVIRONMENT")
+    ; cloud
+    (util/shell! "gcloud" "sql" "databases" "delete" dbname "-i" "zero-postgresql")
+    ; local
+    (util/shell! "dropdb" dbname)))
+
+(defn clean-db-fixture [f]
```

Comment:
very nice.
One point, do you want to export every one of these functions? I'd be tempted to only export the fixture if that's the only one your tests use. (You can make a function namespace priivate using `defn-`)

Reply:
Good point, I'm going to make them private and only expose this fixture. (I don't think we want to turn on this fixture until the integration test is set up in an isolated namespace)